iothub/service/src/Messaging/ServiceClient.cs

```diff
@@ -290,9 +289,7 @@ public virtual async Task OpenAsync()
             if (Logging.IsEnabled)
                 Logging.Enter(this, $"Opening AmqpServiceClient", nameof(OpenAsync));
 
-            using var ctx = new CancellationTokenSource(_openTimeout);
-
-            await _faultTolerantSendingLink.OpenAsync(ctx.Token).ConfigureAwait(false);
+            await _faultTolerantSendingLink.OpenAsync(_openTimeout).ConfigureAwait(false);
```

Comment:
For the `OpenAsync()` method being invoked here, the call stacks are following - passing `TimeSpan` vs `CancellationToken`.
![image](https://github.com/Azure/azure-iot-sdk-csharp/assets/94650966/31471bd9-dcb8-4afb-bb11-18f65d835a6d)
![image](https://github.com/Azure/azure-iot-sdk-csharp/assets/94650966/e3ea43c1-0687-4982-9562-fa0256425fc4)

It's surprising that passing in `TimeSpan` would lead to `TimeSpan.MaxValue` being used in downstream no matter what's the value we are passing. This eventually caused the ArgumentOutOfRangeException.

Reply:
Do we have a cancellation token variant of OpenAsync() in our public API?

Reply:
no, unfortunately we are passing nothing
https://github.com/Azure/azure-iot-sdk-csharp/blob/main/iothub/service/src/Messaging/ServiceClient.cs#L288